### ISSUE TITLE
Compilation error with Criteria

### ISSUE THREAD

**Issue by pascalgrimaud** [jhipster/generator-jhipster#8351]
Following this PR: https://github.com/jhipster/generator-jhipster/pull/8054
It seems to broke all current builds, after v5.3.3 release.

I look Travis, and here the log errors:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 38.714 s
[INFO] Finished at: 2018-09-18T03:34:48+10:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project travis-default: Compilation failure: Compilation failure:
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[194,5] illegal start of expression
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[197,24] ';' expected
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[212,9] illegal start of expression
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[216,27] ';' expected
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[234,2] reached end of file while parsing
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[432,5] illegal start of expression
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[435,24] ';' expected
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[471,9] illegal start of expression
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[475,27] ';' expected
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[514,2] reached end of file while parsing
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
mv: cannot stat ‘target/*.war’: No such file or directory
Error when packaging
```

**Comment by jdubois** [comment]
Yes indeed, so bad news.... I think that the initial bug, that @deepu105 corrected, made that `--with-entities` did not work, so the tests were not run on entities. And so now it works again, we've found some new bugs...
Time to fix and do yet another release :-(

**Comment by gzsombor** [comment]
Sorry, I'm going to fix this, the change looked so innocent, my bad.

**Comment by jdubois** [comment]
Thanks @gzsombor - I'm available right now if needed. I'll wait for your fix to do a new release.

**Comment by pascalgrimaud** [comment]
Thanks guys. I can't work on this any more tonight. I will be able to help tomorrow morning if needed.
Just have the time to reproduce locally and confirm the issue. Here my jhinfo to help:

##### **JHipster Version(s)**

```
jhipster-sample-application@0.0.0 /home/pgrimaud/tmp/15-default
└── (empty)

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "baseName": "jhipsterSampleApplication",
    "packageName": "io.github.jhipster.sample",
    "packageFolder": "io/github/jhipster/sample",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "useSass": true,
    "buildTool": "maven",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "ar-ly",
      "hy",
      "bn",
      "by",
      "ca",
      "zh-cn",
      "zh-tw",
      "cs",
      "da",
      "nl",
      "et",
      "fa",
      "fr",
      "gl",
      "de",
      "el",
      "hi",
      "hu",
      "id",
      "it",
      "ja",
      "ko",
      "mr",
      "pl",
      "pt-br",
      "pt-pt",
      "ro",
      "ru",
      "sk",
      "sr",
      "es",
      "sv",
      "tr",
      "ta",
      "th",
      "ua",
      "uz-lat",
      "vi"
    ],
    "rememberMeKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "testFrameworks": [
      "gatling",
      "protractor"
    ],
    "serverPort": "8080",
    "jhiPrefix": "myPrefix",
    "travis": true,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "clientPackageManager": "npm",
    "clientFramework": "angularX"
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity BankAccount {
  name String required,
  bankNumber Integer,
  agencyNumber Long,
  lastOperationDuration Float,
  meanOperationDuration Double,
  balance BigDecimal required,
  openingDay LocalDate,
  lastOperationDate Instant,
  active Boolean,
  accountType BankAccountType,
  attachment AnyBlob,
  description TextBlob
}
entity Label {
  labelName String required minlength(3)
}
entity Operation {
  date Instant required,
  description String,
  amount BigDecimal required
}

enum BankAccountType {
  CHECKING,
  SAVINGS,
  LOAN
}

relationship OneToMany {
  BankAccount{operation} to Operation{bankAccount(name)}
}
relationship ManyToOne {
  BankAccount{user(login)} to User
}
relationship ManyToMany {
  Operation{label(labelName)} to Label{operation}
}

dto BankAccount with mapstruct
service BankAccount with serviceImpl
angularSuffix BankAccount with mySuffix
filter BankAccount
clientRootFolder BankAccount, Label, Operation with test-root
paginate Label with pagination
service Label with serviceClass
paginate Operation with infinite-scroll

</pre>
</details>


##### **Environment and Tools**

openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-0ubuntu0.18.04.1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)

git version 2.18.0

node: v8.11.4

npm: 6.4.1

yarn: 1.9.4

Docker version 18.03.0-ce, build 0520e24

docker-compose version 1.21.2, build a133471

Congratulations, JHipster execution is complete!

**Comment by jdubois** [comment]
No problem @pascalgrimaud ! I'll do the release tonight - @gzsombor just tell me if you have the time, I can probably also fix this myself

**Comment by gzsombor** [comment]
I have time, but it seems my npm/node/yarn environment went crazy, I fear, I have to setup from ground zero

**Comment by jdubois** [comment]
OK, I'm also having a look. You need Node 8, I had issues with Node 10 recently.

**Comment by gzsombor** [comment]
I have 8.9.4, but it's not recent enough :(

**Comment by gzsombor** [comment]
Thanks, I was much slower, now I see, it is working now.
(Sometimes I forget, what I don't like about npm, but after I have to use it again ... :wink:  )
